contracts-bedrock: deploy config for sepolia

tynes opened the pull request:
**Description**

Adds the deploy config for sepolia. Also clean up a little bit of dead code and parse sepolia L1 and L2 in the deployer script. Optimistic Kovan had a chain id of 69 and is now deprecated, so re-using 69 for Optimistic Sepolia so that 420 can be used by goerli still while both goerli and sepolia run in parallel.

To review the deploy config, the values should almost all match goerli except for 
- the eip 1559 params which should now match mainnet
- The chainids should also be different
- The gas oracle params should match mainnet

<!-- Contributions welcome! See https://github.com/ethereum-optimism/.github/blob/master/CONTRIBUTING.md -->

tynes committed:
contracts-bedrock: deploy config for sepolia

Adds the deploy config for sepolia. Also clean up a little
bit of dead code and parse sepolia L1 and L2 in the deployer
script. Optimistic Kovan had a chain id of 69 and is now deprecated,
so re-using 69 for Optimistic Sepolia so that 420 can be used
by goerli still while both goerli and sepolia run in parallel.

To review the deploy config, the values should all match goerli
except for the eip 1559 params which should now match mainnet.
The chainids should also be different.

protolambda commented on packages/contracts-bedrock/deploy-config/sepolia.json:
```diff
@@ -0,0 +1,41 @@
+  "baseFeeVaultMinimumWithdrawalAmount": "0x8ac7230489e80000",
+  "l1FeeVaultMinimumWithdrawalAmount": "0x8ac7230489e80000",
+  "sequencerFeeVaultMinimumWithdrawalAmount": "0x8ac7230489e80000",
+  "baseFeeVaultWithdrawalNetwork": 0,
+  "l1FeeVaultWithdrawalNetwork": 0,
+  "sequencerFeeVaultWithdrawalNetwork": 0,
```
0 = L1, 1 = L2 is kind of confusing. Seems good otherwise though.

tynes replied:
We could write a go type for handling this, probably useful in the longer term, human friendly string -> solidity type in `UnmarshalJSON`

tynes replied:
@refcell is going to handle writing a go type that marshals `"local"` and `"remote"`

protolambda commented on packages/contracts-bedrock/deploy-config/sepolia.json:
```diff
@@ -0,0 +1,41 @@
+{
+  "finalSystemOwner": "0xBc1233d0C3e6B5d53Ab455cF65A6623F6dCd7e4f",
+  "portalGuardian": "0xBc1233d0C3e6B5d53Ab455cF65A6623F6dCd7e4f",
+  "l1StartingBlockTag": "",
```
Are we picking this explicitly at a later point, or do we just leave it empty?

tynes replied:
We need to explicitly pick this at a later point, after contract deployment
https://github.com/ethereum-optimism/optimism/blob/9af277e9d0ac7cf89d8409234241869241c1b0a3/op-node/cmd/genesis/cmd.go#L136-L140

tynes commented:
The last thing is the L2 chain id, decided to not reuse OP Kovan's

tynes committed:
deploy-config: update values

Set the chain id to be a number that has not been used yet.
Update the batch inbox address so that it corresponds to the
chain id. Also update the gov token owner to be the eoa
that is the final system owner.

tynes committed:
sepolia: update config

tynes committed:
deploy-config: finalize sepolia

tynes committed:
sepolia: deploy L1 contracts

They should also be verified

tynes committed:
foundry: fix config

Ensure that the correct scripts directory is included
so that contract verification can work correctly

tynes committed:
deploy-config: update l1 starting block tag for sepolia

tynes committed:
op-node: fix small bugs in genesis script

tynes committed:
sepolia: restart with 30 million block gas limit

tynes committed:
sepolia: redeploy L1 contracts

K-Ho commented on packages/contracts-bedrock/deploy-config/sepolia.json:
```diff
@@ -0,0 +1,41 @@
+  "l2OutputOracleStartingTimestamp": 0,
+  "l2OutputOracleProposer": "0x02b1786A85Ec3f71fBbBa46507780dB7cF9014f6",
+  "l2OutputOracleChallenger": "0xfd1D2e729aE8eEe2E146c033bf4400fE75284301",
+  "finalizationPeriodSeconds": 12,
```
I like this value for ease of testing withdrawals,but curious if this needs to be a higher value for the testing we want to do internally on FPs

K-Ho commented on packages/contracts-bedrock/scripts/Deployer.sol:
```diff
@@ -445,6 +445,10 @@ abstract contract Deployer is Script {
             return "hardhat";
+        } else if (chainid == 11155111) {
+            return "sepolia";
+        } else if (chainid == 69) {
```
This isn't the same as the chain id above

K-Ho commented on packages/contracts-bedrock/deploy-config/sepolia.json:
```diff
@@ -0,0 +1,41 @@
+  "governanceTokenSymbol": "OP",
+  "governanceTokenName": "Optimism",
+  "governanceTokenOwner": "0xfd1D2e729aE8eEe2E146c033bf4400fE75284301",
+  "l2GenesisBlockGasLimit": "0x17D7840",
```
Should be `0x1C9C380` - 30m max gas limit, elasticity of 6, which gives us a target of 5m gas.

tynes committed:
sepolia: final deploy config

K-Ho commented on packages/contracts-bedrock/deploy-config/sepolia.json:
```diff
@@ -0,0 +1,41 @@
+  "maxSequencerDrift": 600,
+  "sequencerWindowSize": 3600,
+  "channelTimeout": 300,
+  "p2pSequencerAddress": "0x715b7219D986641DF9eFd9C7Ef01218D528e19ec",
```
where does this value come from?

K-Ho commented on packages/contracts-bedrock/deploy-config/sepolia.json:
```diff
@@ -0,0 +1,41 @@
+  "channelTimeout": 300,
+  "p2pSequencerAddress": "0x715b7219D986641DF9eFd9C7Ef01218D528e19ec",
+  "batchInboxAddress": "0xff00000000000000000000000000000011155420",
+  "batchSenderAddress": "0x7431310e026B69BFC676C0013E12A1A11411EEc9",
```
where does this value come from?

tynes committed:
contracts-bedrock: fix deployer chainid mapping

tynes committed:
sepolia: final deployment

tynes committed:
sepolia: starting l1 block tag
